x-pack/plugins/case/server/routes/api/utils.ts

```diff
@@ -111,55 +112,50 @@ export const getAlertIds = (comment: CommentRequest): string[] => {
   return [];
 };
 
-/**
- * This structure holds the alert IDs and indices found from multiple alert comments
- */
-export interface AlertInfo {
-  ids: string[];
-  indices: Set<string>;
-}
-
-const accumulateIndicesAndIDs = (comment: CommentAttributes, acc: AlertInfo): AlertInfo => {
-  if (isCommentRequestTypeAlertOrGenAlert(comment)) {
-    acc.ids.push(...getAlertIds(comment));
-    const indices = Array.isArray(comment.index) ? comment.index : [comment.index];
-    indices.forEach((index) => acc.indices.add(index));
-  }
-  return acc;
+const getIDsAndIndicesAsArrays = (
+  comment: CommentRequestAlertType
+): { ids: string[]; indices: string[] } => {
+  return {
+    ids: Array.isArray(comment.alertId) ? comment.alertId : [comment.alertId],
+    indices: Array.isArray(comment.index) ? comment.index : [comment.index],
+  };
 };
 
 /**
- * Builds an AlertInfo object accumulating the alert IDs and indices for the passed in alerts.
+ * This functions extracts the ids and indices from an alert comment. It enforces that the alertId and index are either
+ * both strings or string arrays that are the same length. If they are arrays they represent a 1-to-1 mapping of
+ * id existing in an index at each position in the array. This is not ideal. Ideally an alert comment request would
+ * accept an array of objects like this: Array<{id: string; index: string; ruleName: string ruleID: string}> instead.
+ *
+ * To reformat the alert comment request requires a migration and a breaking API change.
```

Comment:
Can you create an issue for this when you have time, so we can discuss it there? It sounds like it's worth making the change for more stability. And the changed API seems much easier to potentially work with

Reply:
Yeah good idea. I'll create the issue 👍

Reply:
https://github.com/elastic/security-team/issues/889

Reply:
Thanks!